test/test_turbulence_closures.jl

```diff
@@ -0,0 +1,144 @@
+function test_closure_instantiation(; closures=(
+        :ConstantSmagorinsky,
+        :ConstantIsotropicDiffusivity,
+        :DirectionalDiffusivity)
+        )
+
+    results = Bool[]
+
+    for closurename in closures
+        closure64 = getproperty(TurbulenceClosures, closurename)()
+        closure32 = getproperty(TurbulenceClosures, closurename)(Float32)
+
+        push!(results, eltype(closure64) == Float64)
+        push!(results, eltype(closure32) == Float32)
+    end
+
+    return all(results)
+end
+
+function test_function_interpolation()
+    grid = RegularCartesianGrid((3, 3, 3), (3, 3, 3))
+    ϕ = rand(3, 3, 3)
+    ϕ² = ϕ.^2
+
+    ▶x_ϕ_c = (ϕ²[2, 2, 2] + ϕ²[1, 2, 2]) / 2
+    ▶x_ϕ_f = (ϕ²[3, 2, 2] + ϕ²[2, 2, 2]) / 2
+
+    ▶y_ϕ_c = (ϕ²[2, 2, 2] + ϕ²[2, 1, 2]) / 2
+    ▶y_ϕ_f = (ϕ²[2, 3, 2] + ϕ²[2, 2, 2]) / 2
+
+    ▶z_ϕ_c = (ϕ²[2, 2, 2] + ϕ²[2, 2, 1]) / 2
+    ▶z_ϕ_f = (ϕ²[2, 2, 3] + ϕ²[2, 2, 2]) / 2
+
+    f(i, j, k, grid, ϕ) = ϕ[i, j, k]^2
+
+    return (
+        TurbulenceClosures.▶x_caa(2, 2, 2, grid, f, ϕ) == ▶x_ϕ_c &&
```

Comment:
Might be nice to import these functions at the top to avoid having to repeat `TurbulenceClosures.`